components/cd-disclosure/cd-disclosure.js

```diff
@@ -0,0 +1,81 @@
+(function (Drupal) {
+  'use strict';
+
+  /**
+   * @see https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/
+   */
+  Drupal.behaviors.cdDisclosure = {
+    idPrefix: 'cd-disclosure--',
+
+    attach: function (context, settings) {
+      // Initialize toggable dropdown.
+      this.initializeDisclosures();
+    },
+
+    /**
+     * Set up all Disclosure widgets found in the DOM.
+     *
+     * For our purposes, the `disclosure` element is the wrapped content itself
+     * and the button to toggle it is generated automatically by this script.
+     */
+    initializeDisclosures: function () {
+      let disclosures = document.querySelectorAll('.cd-disclosure');
```

Comment:
I actually started out with data attribute only, but switched because one thing that all components in the CD have is that they have a top-level CSS classname, including CD Dropdown. That's why I landed on the solution as written. But I do agree that in terms of how JS functions, using the attribute makes sense.

I've added the data-attr back in, and keep the optional class in the example. The JS will look for the attribute, not the class.

If we ever offered some styling variant, or for example wanted to offer declarative transitions like slide-up/down, then I think we can agree that modifier classes are the established standard, both in our CD codebase and as a general industry convention.